src/manifolds/ProductManifold.jl

```diff
@@ -248,8 +226,8 @@ function flat!(M::ProductManifold, ξ::CoTFVector, p, X::TFVector)
 end
 
 function get_basis(M::ProductManifold, p, B::AbstractBasis)
-    parts = map(t -> get_basis(t..., B), ziptuples(M.manifolds, submanifold_components(p)))
-    return PrecomputedProductOrthonormalBasis(parts)
+    parts = map(t -> get_basis(t..., B).parts, ziptuples(M.manifolds, submanifold_components(p)))
```

Comment:
Why did you add this `.parts` thing here? We should store here either the entire basis or its `.data` field perhaps.

Reply:
The parts themselves are `CachedBases` I think? I was trying to understand what it does and this yields consistent results, namely `get_basis` delivers a `CachedBasis` with clever internal storage that the other three functions work with. I hope.

Reply:
> The parts themselves are `CachedBases` I think?

Yes, and `CachedBasis` has only one field called `data`. There is no `.parts` field. This should work without the `.parts` thing.

Reply:
Interesting, must have also slipped in my slight confusion (see above).